wolfcrypt/src/random.c

```diff
@@ -1770,6 +1770,24 @@ WC_RNG* wc_rng_new(byte* nonce, word32 nonceSz, void* heap)
 }
 
 
+WOLFSSL_ABI
+WC_RNG* wc_rng_new_ex(byte* nonce, word32 nonceSz, void* heap, int devId)
+{
+    WC_RNG* rng;
+
+    rng = (WC_RNG*)XMALLOC(sizeof(WC_RNG), heap, DYNAMIC_TYPE_RNG);
+    if (rng) {
+        int error = _InitRng(rng, nonce, nonceSz, heap, devId) != 0;
+        if (error) {
```

Comment:
I think it's probably a mistake to double down on the blinded-error-code oversight in `wc_rng_new()`.  This is the `_ex` version so this is an opportunity to fix it, with something like
```
int wc_rng_new_ex(WC_RNG **rng, byte* nonce, word32 nonceSz, void* heap, int devId) {
    int ret;
    *rng = (WC_RNG*)XMALLOC(sizeof(WC_RNG), heap, DYNAMIC_TYPE_RNG);
    [...]
    return ret;
}
```
Also `error` shouldn't be used here because it collides with a glibc function.

Reply:
@douzzer addressed API changes in latest commit, lmk what you think